activerecord/lib/active_record/inheritance.rb

```diff
@@ -50,8 +50,10 @@ def new(*args, &block)
           raise NotImplementedError, "#{self} is an abstract class and cannot be instantiated."
         end
 
-        attrs = args.first
+        attrs = args.first || {}
         if subclass_from_attributes?(attrs)
+          attrs = attrs.with_indifferent_access
+          attrs[inheritance_column] ||= columns_hash[inheritance_column].default
```

Comment:
@kuldeepaggarwal shouldn't this happen before `if subclass_from_attributes?(attrs)`? Otherwise `subclass_from_attributes?(attrs)` will not see the default.